packages/vite/src/node/plugins/css.ts

```diff
@@ -1234,14 +1236,41 @@ const UrlRewritePostcssPlugin: PostCSS.PluginCreator<{
           promises.push(
             rewriterToUse(declaration.value, replacerForDeclaration).then(
               (url) => {
-                declaration.value = url
+                let match
+
+                if (
+                  isCssFile &&
+                  (match = url.match(/url\(['"]?data:.+?;base64,.+?\)/))
+                ) {
+                  const [base64] = match
+
```

Comment:
I think the map can also reduce the hash calc.
```ts
let varname!: string
if(rootVars[base64]) {
  varname = rootVars[base64]
}. else {
  varname = `--base64-${getHash(base64)}`
}
```

Reply:
Good catch. I think we can use only one `WeakMap` to store all the base64 variables across all the CSS files